src/screens/Account.jsx

```diff
@@ -4,17 +4,19 @@ import CandidateForm from "../components/CandidateForm";
 import PageTitle from "../components/PageTitle";
 import Loader from "../components/Loader";
 import { AppContext } from "../Context"
+import { editCandidate } from "../utils/request"
 
 const Account = () => {
   const [fields, setFields] = useState(null);
-  const { user } = useContext(AppContext)
+  const { user, setUser } = useContext(AppContext)
 
   useEffect(() => {
     setFields(Object.keys(user.personal).map(item => {return {name: item, value: user.personal[item]}}))
   }, []);
 
-  const onFormSubmit = (values) => {
-    console.log(values);
+  const onFormSubmit = async (values) => {
+    const hasChanged = await editCandidate(values, user.id)
+    hasChanged && setUser({...user, personal: values})
```

Comment:
Really good. in here you could have gone 2 ways.

Either do it as you are, updating the global user with the "cached" values that you have in state, or create a function in App.js that makes a call to /me, which you would pass inside the Account component, and you would call it everytime you update the user. 

But your approach is really good, you would use the other approach when a change in the Account form for example would change some other properties in the user, like the id, role, etc. Then you would have to fetch fresh data from the server.

Reply:
My thoughts exactly :)